generator/src/main/java/com/scottlogic/deg/generator/ViolationGenerationEngine.java

```diff
@@ -47,12 +48,19 @@ public void generateDataSet(Profile profile, GenerationConfig config, FileOutput
         catch (Exception e){}
 
         for (ViolatedProfile violated: violatedProfiles) {
-            generationEngine.generateDataSet(violated, config,
-                fileOutputTarget.withFilename(intFormatter.format(filename)));
+            standardGenerationEngine.generateDataSet(violated, config,
+                getOutputTargetWithFilename(outputTarget, intFormatter.format(filename)));
             filename++;
         }
+    }
 
-
+    private OutputTarget getOutputTargetWithFilename(OutputTarget outputTarget,  String filename) {
+        if (outputTarget.getClass() == FileOutputTarget.class) {
```

Comment:
```suggestion
        if (outputTarget instanceOf FileOutputTarget) {
```

Reply:
Not sure about this, but may be a necessary evil.

Reply:
Needs to be some evil - either this class deals with the difference between an OutputTarget and a FileOutputTarget or the interface OutputTarget has to include a method that's only relevant for one implementation of the interface.